lte/cloud/go/services/subscriberdb_cache/worker.go

```diff
@@ -16,28 +16,28 @@ package subscriberdb_cache
 import (
 	"time"
 
-	lte_protos "magma/lte/cloud/go/protos"
 	lte_models "magma/lte/cloud/go/services/lte/obsidian/models"
 	"magma/lte/cloud/go/services/subscriberdb"
-	"magma/lte/cloud/go/services/subscriberdb/storage"
 	"magma/orc8r/cloud/go/clock"
 	"magma/orc8r/cloud/go/services/configurator"
+	"magma/orc8r/cloud/go/syncstore"
+	"magma/orc8r/lib/go/protos"
 
 	"github.com/golang/glog"
 	"github.com/hashicorp/go-multierror"
 	"github.com/pkg/errors"
 	"github.com/thoas/go-funk"
 )
 
-func MonitorDigests(config Config, digestStore storage.DigestStore, perSubDigestStore *storage.PerSubDigestStore, subStore *storage.SubStore) {
+func MonitorDigests(config Config, store syncstore.SyncStore) {
```

Comment:
So what's the verdict: you think worker.go can be at least somewhat generalized? If yes I'll have my next intern work on it 🙂

Reply:
I think definitely with the use of a worker object that accepts custom functions as initializing arguments. Currently, the `subscriberdb` worker follows the workflow of
- Every <loop_interval>, Initiate loop where
  - garbage collection (takes in `trackedNetworks`, rest is left to `syncstore`)
  - find networks to renew (takes in `trackedNetworks`, rest depends on networks in `syncstore`)
  - renew digests (uses custom function to generate new digest trees)
  - renew cached objects (uses custom function to generate new cached objects)

All these steps are generic enough to be propagated for other service workers as well, as long as we take in the service functions to generate new digests & cached objects @hcgatewood